### PR TITLE
feat: in_range filtering

### PR DESCRIPTION
**Pull request by vgilabert94**
#### Changes
Following cv2.inRange [OpenCV function](https://docs.opencv.org/3.4/d2/de8/group__core__array.html#ga48af0ab51e36436c5d04340e036ce981), I've implemented a pytorch version for batched images. 
The function needs the image, and two bounds for filtering (lower and upper). 
There is the possibility to add a parameter (return_mask) to return the mask and otherwise the filtered image.

Code to get image:
```
BASE_IMAGE_URL = "https://raw.githubusercontent.com/kornia/data/main/panda.jpg"
dev = 'cpu'
img = read_img_from_url(BASE_IMAGE_URL, (256, 256))
img_hsv = K.color.rgb_to_hsv(img)
h, s, v = torch.split(img_hsv, split_size_or_sections=1, dim=1)
h = h / (2*torch.pi)
img_hsv = torch.cat((h, s, v), dim=1)
img_hsv = img_hsv.to(dev)
img = img.to(dev)
lower = (0.314, 0.2, 0.2)
upper = (0.47, 1.0, 1.0)
lower = torch.tensor(lower).reshape(1,3,1,1)
upper = torch.tensor(upper).reshape(1,3,1,1)
mask = in_range(img_hsv, lower, upper)
out = img * mask
```
image

#### Type of change

- [ ] 📚  Documentation Update
- [ ] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Review by johnnv1 — APPROVED** [review]
nice patch. lgtm, only needs the green lights on CI

**Comment by vgilabert94** [comment]
> nice patch. lgtm, only needs the green lights on CI

I need help to solve this errors. I don't know how to debug properly without doing commit and push every time.

File: tests/filters/test_in_range.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+import torch
+
+from kornia.filters import InRange, in_range
+
+from testing.base import BaseTester, assert_close
+
+
+def test_in_range(device, dtype):
+    torch.manual_seed(1)
+    input_tensor = torch.rand(1, 3, 3, 3, device=device)
```
Comment (edgarriba) [inline tests/filters/test_in_range.py:13]:
```suggestion
    input_tensor = torch.rand(1, 3, 3, 3, device=device, dtype=dtype)
```

**Review by edgarriba — APPROVED** [review]

**Comment by johnnv1** [comment]
> > nice patch. lgtm, only needs the green lights on CI
> 
> I need help to solve this errors. I don't know how to debug properly without doing commit and push every time.

if it's some doc issue, you can check with make build-doc or make doctest... for testing falling cross OS, I would go with commit and push too

**Comment by johnnv1** [comment]
@vgilabert94  could you rebase this one? i removed the CI with the issue

**Comment by vgilabert94** [comment]
> @vgilabert94 i guess something is missing with the docstring format -- i suggest to run `make build-docs` and `make doctest`

`make doctest` OK
`make build-docs` can't run in local. I'm on windows:
```
PS C:\Users\LENOVO\REPOSITORIOS\kornia> make build-docs 
cd docs; make clean html
El sistema no puede encontrar la ruta especificada.
make: *** [Makefile:52: build-docs] Error 1
```

**Comment by edgarriba** [comment]
ok, i'll handle it
